clippy_lints/src/no_effect.rs

```diff
@@ -153,11 +162,26 @@ fn check_no_effect(cx: &LateContext<'_>, stmt: &Stmt<'_>) -> bool {
     false
 }
 
-fn has_no_effect(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
-    if expr.span.from_expansion() {
-        return false;
+fn is_operator_overrided(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
```

Comment:
```suggestion
fn is_operator_overriden(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
```

Reply:
I have added check for unary operator and fix my typo, test for unary operator is added to `ui/test/no_effect.*` also.
It's nice of you to give a lot reliable suggestions.
Ready to be merged !